electrum_nmc/electrum/names.py

```diff
@@ -1152,11 +1182,16 @@ def get_domain_records_map(domain, value):
 def add_domain_record(base_domain, value, record):
     domain, record_type, data = record
 
-    # Handle Tor records specially
-    if record_type == "address" and data[0] == "tor":
-        domain = "_tor." + domain
-        record_type = "txt"
-        data = data[1]
+    # Handle Tor, IPFS & IPNS records specially
+    if record_type == "address":
+        if data[0] == "tor":
+            domain = "_tor." + domain
+            record_type = "txt"
+            data = data[1]
+        elif data[0] in ["ipfs", "ipns"]:
+            domain = "_dnslink." + domain
+            record_type = "txt"
+            data = f"dnslink=/{data[0]}/{data[1]}" # To prevent duplicating the prefix, remove the prefix when importing the record (e.g in insert_record function in GUI(qt) ) 
```

Comment:
We can add an if statement to avoid duplicating the prefix (This is already handled in QT) Maybe for extra safety